### PR TITLE
Try to fix segmentation fault related to last updates in cuDNN conv codes.

### PR DESCRIPTION
**Pull request by notoraptor**
@nouiz @abergeron

### CODE REVIEW

File: theano/gof/cc.py [hunk 1]
```diff
@@ -235,6 +235,7 @@ def struct_gen(args, struct_builders, blocks, sub):
 
     # declares the storage
     storage_decl = "\n".join(["PyObject* %s;" % arg for arg in args])
+    storage_init = "\n".join(["this->%s = NULL;" % arg for arg in args])
```
Comment (abergeron) [inline theano/gof/cc.py]: We don't need this, there is no problem with the storage.

Comment (nouiz) [inline theano/gof/cc.py]: Does this remove the need for the memset? I don't think so it enough to remove the memset, as it won't fix other case we cover, but it still greatly help. Op that don't have there own struct_decl (they just use params) won't need the memset.

Comment (nouiz) [inline theano/gof/cc.py]: Forget my comments, it is for the storage, not for the param. See @abergeron comments. I'll delete mine.

Comment (notoraptor) [inline theano/gof/cc.py]: Indeed, removed.

**Comment by nouiz** [comment]
2 person told me it fix there problem, so we can merge when jenkins CI pass. We don't need to wait for travis CI.

**Pull request merged**